tools/syz-cover/syz-cover.go

```diff
@@ -156,65 +156,55 @@ func main() {
 	}
 	pcs := initPCs(rg)
 	progs := []cover.Prog{{PCs: pcs}}
-	buf := new(bytes.Buffer)
 	params := cover.HandlerParams{
 		Progs: progs,
 	}
-	if *flagExportCSV != "" {
-		if err := rg.DoCSV(buf, params); err != nil {
-			tool.Fail(err)
-		}
-		if err := osutil.WriteFile(*flagExportCSV, buf.Bytes()); err != nil {
-			tool.Fail(err)
-		}
-		return
-	}
+
 	if *flagExportLineJSON != "" {
-		if err := rg.DoLineJSON(buf, params); err != nil {
-			tool.Fail(err)
-		}
-		if err := osutil.WriteFile(*flagExportLineJSON, buf.Bytes()); err != nil {
-			tool.Fail(err)
-		}
+		doReport(params, *flagExportLineJSON, rg.DoLineJSON)
 		return
 	}
 	if *flagExportJSONL != "" {
-		if err := rg.DoCoverJSONL(buf, params); err != nil {
-			tool.Fail(err)
-		}
-		if err := osutil.WriteFile(*flagExportJSONL, buf.Bytes()); err != nil {
-			tool.Fail(err)
-		}
-		return
-	}
-	if err := rg.DoHTML(buf, params); err != nil {
-		tool.Fail(err)
-	}
-	if *flagExportHTML != "" {
-		if err := osutil.WriteFile(*flagExportHTML, buf.Bytes()); err != nil {
-			tool.Fail(err)
-		}
+		doReport(params, *flagExportJSONL, rg.DoCoverJSONL)
 		return
 	}
-	fn, err := osutil.TempFile("syz-cover")
-	if err != nil {
+
+	log.Logf(1, "start DoHTML")
```

Comment:
I think it's a bit of an overkill to always generate that many programs and start that many gui processes. It's better to keep the list of generated reports configurable.

Reply:
added -exports option.